### PR TITLE
NEWS for to_indices (#19730)

### PR DESCRIPTION
**Pull request by mbauman**
Cc @Sacha0

### CODE REVIEW

**Comment by Keno** [comment]
Should target 0.6?

**Review by Sacha0 — COMMENTED** [review]
This is great! Thanks @mbuman! :)

Perhaps the specific breaking changes and deprecations are worth mentioning as well? And running `NEWS-update.jl` to link the PR?

**Comment by tkelman** [comment]
> Should target 0.6?

yep, this is currently adding to the 0.7 section of the file, should move it down to the 0.6 section

**Comment by mbauman** [comment]
I've added a few sub-points that help explain the changes a bit more in depth.

File: NEWS.md [hunk 1]
```diff
@@ -404,7 +404,17 @@ Library improvements
...
     for any unsupported indices, and the returned arrays should be iterated over (and
-    not indexed into) to support more efficient linear indexing ([#19730]).
+    not indexed into) to support more efficient logical indexing ([#19730]).
+
+    + Using colons (`:`) to represent a collection of indices is deprecated. They now must be
+      explicitly converted to a specialized array of integers with the `to_indices` function.
+      As a result, the type of `SubArray`s that represent views over colon indices has changed.
+
+    + Logical indexing is now more efficient. Logical arrays are converted by `to_indices` to
+      a lazy, iterative collection of indices that doesn't support indexing. A deprecation
```
Comment (tkelman) [inline NEWS.md]: iterative or iterable?

Comment (mbauman) [inline NEWS.md]: thanks

**Review by Sacha0 — APPROVED** [review]
Fantastic! Thanks @mbauman! :)

**Comment by Sacha0** [comment]
Absent objections, requests for time, or someone beating me to it, I plan to merge this tomorrow morning. Best!

**Comment by Sacha0** [comment]
(Ref. #21475.)

**Pull request merged**